components/places/src/error.rs

```diff
@@ -4,78 +4,76 @@
-    #[fail(display = "Invalid place info: {}", _0)]
+    #[error("Invalid place info: {0}")]
     InvalidPlaceInfo(InvalidPlaceInfo),
 
-    #[fail(display = "The store is corrupt: {}", _0)]
+    #[error("The store is corrupt: {0}")]
     Corruption(Corruption),
 
-    #[fail(display = "Error synchronizing: {}", _0)]
-    SyncAdapterError(#[fail(cause)] sync15::Error),
+    #[error("Error synchronizing: {0}")]
+    SyncAdapterError(#[from] sync15::Error),
 
-    #[fail(display = "Error merging: {}", _0)]
-    MergeError(#[fail(cause)] dogear::Error),
+    #[error("Error merging: {0}")]
+    MergeError(#[from] dogear::Error),
 
-    #[fail(display = "Error parsing JSON data: {}", _0)]
-    JsonError(#[fail(cause)] serde_json::Error),
+    #[error("Error parsing JSON data: {0}")]
+    JsonError(#[from] serde_json::Error),
 
-    #[fail(display = "Error executing SQL: {}", _0)]
-    SqlError(#[fail(cause)] rusqlite::Error),
+    #[error("Error executing SQL: {0}")]
+    SqlError(#[from] rusqlite::Error),
 
-    #[fail(display = "Error parsing URL: {}", _0)]
-    UrlParseError(#[fail(cause)] url::ParseError),
+    #[error("Error parsing URL: {0}")]
+    UrlParseError(#[from] url::ParseError),
 
-    #[fail(display = "A connection of this type is already open")]
+    #[error("A connection of this type is already open")]
     ConnectionAlreadyOpen,
 
-    #[fail(display = "An invalid connection type was specified")]
+    #[error("An invalid connection type was specified")]
     InvalidConnectionType,
 
-    #[fail(display = "IO error: {}", _0)]
-    IoError(#[fail(cause)] std::io::Error),
+    #[error("IO error: {0}")]
+    IoError(#[from] std::io::Error),
 
-    #[fail(display = "Operation interrupted")]
-    InterruptedError(#[fail(cause)] Interrupted),
+    #[error("Operation interrupted")]
+    InterruptedError(#[from] Interrupted),
 
-    #[fail(display = "Tried to close connection on wrong PlacesApi instance")]
+    #[error("Tried to close connection on wrong PlacesApi instance")]
     WrongApiForClose,
 
-    #[fail(display = "Incoming bookmark missing type")]
+    #[error("Incoming bookmark missing type")]
     MissingBookmarkKind,
 
-    #[fail(display = "Incoming bookmark has unsupported type {}", _0)]
+    #[error("Incoming bookmark has unsupported type {0}")]
     UnsupportedIncomingBookmarkType(JsonValue),
 
-    #[fail(display = "Synced bookmark has unsupported kind {}", _0)]
+    #[error("Synced bookmark has unsupported kind {0}")]
     UnsupportedSyncedBookmarkKind(u8),
 
-    #[fail(display = "Synced bookmark has unsupported validity {}", _0)]
+    #[error("Synced bookmark has unsupported validity {0}")]
     UnsupportedSyncedBookmarkValidity(u8),
 
     // This will happen if you provide something absurd like
     // "/" or "" as your database path. For more subtley broken paths,
     // we'll likely return an IoError.
-    #[fail(display = "Illegal database path: {:?}", _0)]
+    #[error("Illegal database path: {0:?}")]
```

Comment:
this is an interesting way to format the error arg, does that work?

Reply:
According to https://github.com/dtolnay/thiserror/blob/master/README.md it translates to `write!("{:?}", self.0)`

Reply:
cool!